Multi-site extension

hubermat opened the pull request:
Konfiguration der Sites über das Script changeConfig.js oder von Hand in der ctldap.config

hubermat committed:
add multisite configuration

hubermat committed:
add multisite config to example config file

hubermat committed:
Merge branch 'multisite'

hubermat committed:
add config to convert email to lowercase

hubermat committed:
Merge branch 'email-lowercase'

hubermat committed:
add script to change config

hubermat committed:
change output

hubermat committed:
fix output of changeConfig.js

hubermat committed:
add hashing of ldap_password, fix caching bug

hubermat committed:
autodetect hashed password

hubermat committed:
add comment option

hubermat committed:
remove error objects in console.log

hubermat committed:
only output the error message, not the whole error object

hubermat committed:
add list command

hubermat committed:
fix bugs

sscholl commented:
Das scheinen ja ein paar Anpassungen zu sein. Was tut der PR alles?

hubermat commented:
Der PR erweitert den ctldap dahingehend, daß Daten von mehreren ChurchTools-Installationen mit einem ctldap abgeholt werden können - aber jeweils in getrennten Namensräumen, wie wenn ctldap x-Mal laufen würde. Also getrennter Login und getrennter Abruf und Cache pro ChurchTools-Installation.
Die Zugänge zu den ChurchTools-Installationen werden in den ctldap.config (siehe Example config) oder über das Config-Script eingerichtet. Das Config-Script modifiziert die ctldap.config entsprechend.

Anstelle den ChurchTools-User für den Datenabruf mit E-Mail und Passwort anzugeben, kann auch das gehashte Passwort aus der ChurchTools-Datenbank eingetragen werden. Dann steht kein Klartext-Passwort mehr in der Config. Dies wird durch die Konfigurationsoption ```ldap_password_bcrypt```aktiviert.

Eine weitere Konfigurationsoption ```email_lower_case``` bewirkt, daß alle E-Mail-Adressen automatisch nach lowercase convertiert werden. Manche LDAP-Clients (z.B. Seafile) machen Probleme, wenn E-Mail-Adressen der User in ChurchTools Großbuchstaben enthalten.

Die Konfigurationsoptionen können global oder auch pro ChurchTools-Installation eingestellt werden.

Alle diese Möglichkeiten sind komplett optional. Wenn nichts weiter konfiguriert wird, verhält sich ctldap weiterhin genau wie bisher (abwärtskompatibel).

sscholl commented:
Das klingt toll. Ich habe nicht den kompletten Code gereviewed. Sieht aber soweit gut aus.

hubermat committed:
prevent brute force attacks

a-schild committed:
Corrected typo in comment

a-schild committed:
Unify debug logging to simplify search

a-schild committed:
Added log level to other log messages

a-schild committed:
Corretcly map ct strasse to ldap street

a-schild committed:
- Upgraded to ldapjs 1.0.2
- substring queries are now case insensitive
  Was an issue in in nextcloud group sharing for example

a-schild committed:
Fixed login problem for ldap users
Service script changed version

hubermat committed:
Merge pull request #1 from a-schild/master

Correct mapping of street field, Cases insensitive group search

a-schild committed:
Merge pull request #1 from churchtools/master

Merge pull request #1 from a-schild/master

a-schild committed:
Add IP to listen for ldap requests, default 127.0.0.1 for localhost only
Detect iptables port forwarding correctly, only when not commented out in config file

hubermat committed:
Merge pull request #2 from a-schild/master

Enhanced security settings and better iptables handling

hubermat committed:
add filter for duplicate emails

hubermat committed:
Merge branch 'master' of https://github.com/churchtools/ctldap-ms

hubermat committed:
fix bug

hubermat committed:
add csrf token

hubermat committed:
fix error handling, improve CSRF handling, improve debug output

milux commented:
Manually merged into develop, closing.
